unitary/examples/quantum_chinese_chess/board.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2023 The Unitary Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unitary.alpha as alpha
+from unitary.examples.quantum_chinese_chess.enums import (
+    SquareState,
+    Color,
+    Type,
+    Language,
+)
+from unitary.examples.quantum_chinese_chess.piece import Piece
+
+
+# The default initial state of the game.
+_INITIAL_FEN = "RHEAKAEHR/9/1C5C1/P1P1P1P1P/9/9/p1p1p1p1p/1c5c1/9/rheakaehr w---1"
+
+
+class Board:
+    """Board holds the assemble of all pieces. Each piece could be either in classical or quantum state."""
+
+    def __init__(self, fen: str = _INITIAL_FEN):
+        self.king_locations = []
+        self.load_fen(fen)
+
+    def load_fen(self, fen: str):
```

Comment:
What do you think about doing something like this instead?

```
def __init__(self, board: alpha.QuantumWorld, current_player: int):
    ....

@classmethod
def from_fen(cls, fen: str) -> Board:
   ...
  return cls(board, current_player)
```

Reply:
Thanks Doug. It's updated.